domain/extract.go

```diff
@@ -0,0 +1,351 @@
+	}
+	records, err := w.collectRecords(ctx, task)
+	if err != nil {
+		logutil.BgLogger().Error("collect stmt summary records failed for extract plan task", zap.Error(err))
+		return "", err
+	}
+	p := w.packageExtractPlanRecords(records)
+	return w.dumpExtractPlanPackage(p)
+}
+
+func (w *extractWorker) collectRecords(ctx context.Context, task *ExtractTask) (map[stmtSummaryHistoryKey]stmtSummaryHistoryRecord, error) {
+	exec := w.sctx.(sqlexec.RestrictedSQLExecutor)
+	ctx1 := kv.WithInternalSourceType(ctx, kv.InternalTxnStats)
+	rows, _, err := exec.ExecRestrictedSQL(ctx1, nil, fmt.Sprintf("SELECT STMT_TYPE, SCHEMA_NAME, TABLE_NAMES, DIGEST, PLAN_DIGEST,QUERY_SAMPLE_TEXT FROM INFORMATION_SCHEMA.STATEMENTS_SUMMARY_HISTORY WHERE SUMMARY_END_TIME > '%s' OR SUMMARY_BEGIN_TIME < '%s'",
+		task.Begin.Format(types.TimeFormat), task.End.Format(types.TimeFormat)))
+	if err != nil {
+		return nil, err
+	}
+	collectMap := make(map[stmtSummaryHistoryKey]stmtSummaryHistoryRecord, 0)
+	is := GetDomain(w.sctx).InfoSchema()
+	for _, row := range rows {
+		record := stmtSummaryHistoryRecord{}
+		record.stmtType = row.GetString(0)
+		record.schemaName = row.GetString(1)
+		record.digest = row.GetString(3)
+		record.planDigest = row.GetString(4)
+		record.sql = row.GetString(5)
+		if !checkRecordValid(record) {
+			continue
+		}
+		key := stmtSummaryHistoryKey{
+			digest:     record.digest,
+			planDigest: record.planDigest,
+		}
+		record.tables = make([]tableNamePair, 0)
+		tables := row.GetString(2)
+		setRecord := true
+
+		for _, t := range strings.Split(tables, ",") {
+			names := strings.Split(t, ".")
+			if len(names) != 2 {
+				setRecord = false
+				break
+			}
+			dbName := names[0]
+			tblName := names[1]
+			t, err := is.TableByName(model.NewCIStr(dbName), model.NewCIStr(tblName))
+			if err != nil {
+				return nil, err
+			}
+			record.tables = append(record.tables, tableNamePair{DBName: dbName, TableName: tblName, IsView: t.Meta().IsView()})
+		}
+		if setRecord {
+			collectMap[key] = record
+		}
+	}
+	return collectMap, nil
+}
+
+func checkRecordValid(r stmtSummaryHistoryRecord) bool {
+	if r.schemaName == "" {
+		return false
+	}
+	if r.planDigest == "" {
+		return false
+	}
+	return true
+}
+
+func (w *extractWorker) packageExtractPlanRecords(records map[stmtSummaryHistoryKey]stmtSummaryHistoryRecord) *extractPlanPackage {
+	p := &extractPlanPackage{}
+	p.sqls = make([]string, 0)
+	p.skippedSQLs = make([]string, 0)
+	p.tables = make(map[tableNamePair]struct{}, 0)
+	for _, record := range records {
+		if strings.Contains(record.sql, "(len:") {
+			p.skippedSQLs = append(p.skippedSQLs, record.sql)
+			continue
+		}
+		// TODO: skip internal schema record
+
+		p.sqls = append(p.sqls, record.sql)
+		for _, tbl := range record.tables {
+			p.tables[tbl] = struct{}{}
+		}
+	}
+	return p
+}
+
+func (w *extractWorker) dumpExtractPlanPackage(p *extractPlanPackage) (name string, err error) {
```

Comment:
Better to add some comments for this function to explain how we organize the package, just like `DumpPlanReplayerInfo`.